### ISSUE TITLE
[Known Issue] LinkedAuthorizationFailed with AKS clusters

### ISSUE THREAD

**Issue by aramase** [Azure/aad-pod-identity#753]
The AKS clusters have switched to using a different os base image as part of the last rollout.

tl;dr **Update to aad-pod-identity version 1.6+**

Releases of `aad-pod-identity` prior to `1.6` used `CreateOrUpdate` API to assign/un-assign identities. This API call requires additional permissions on the new base image subscription and as a result will fail with error -

```
E0821 10:42:19.340799       1 mic.go:799] Updating msis on node aks-xxxxxxx-21457229-vmss, add [1], del [0] failed with error compute.VirtualMachineScaleSetsClient#CreateOrUpdate: Failure sending request: StatusCode=403 -- Original Error: Code="LinkedAuthorizationFailed" Message="The client has permission to perform action 'Microsoft.Compute/galleries/images/versions/read' on scope '/subscriptions/<subid>/resourceGroups/<rg>/providers/Microsoft.Compute/virtualMachineScaleSets/aks-nodepool-vmss', however the current tenant 'xxxx' is not authorized to access linked subscription '109a5e88-712a-48ae-9078-9ca8b3c81345'."
```

With 1.6+ releases of `aad-pod-identity`, the identity is assigned/un-assigned using [PATCH API calls](https://github.com/Azure/aad-pod-identity/pull/522) which do not require these additional permissions on the base image. 

AKS has already sent out a note to all clusters still running old version of `aad-pod-identity` requesting upgrade to 1.6+ releases.

**Note: Please review this doc - [v1.6+ breaking change](https://github.com/Azure/aad-pod-identity#v160-breaking-change) as part of the upgrade**

**Comment by aullom1** [comment]
What do I do if I get the error

```
The client has permission to perform action 'Microsoft.Compute/galleries/images/versions/read' on scope '/subscriptions/xxxxxx/resourceGroups/mc_rg-xxxxx/providers/Microsoft.Compute/virtualMachines/aks-agentpool-xxxxx-0', however the current tenant 'xxxxxxx' is not authorized to access linked subscription '109a5e88-712a-48ae-9078-9ca8b3c81345'.
```

on newly deployed clusters, but I can't upgrade to 1.6 due to enterprise control of the aad-pod-identity version?  Also, note that this is _not_ VMSS

**Comment by chewong** [comment]
@aullom1 Unfortunately you will need to upgrade pod identity to the latest releaese, which uses `PATCH` to assign / unassign identity from the underling VM / VMSS instead of CreateOrUpdate, which should get rid of the error.

**Comment by leeeennyy** [comment]
Hi @aramase , do you know which base images for AKS VMSS will AAD Pod Identity < 1.6+ have trouble with?

**Comment by aullom1** [comment]
@leeeennyy I can't say for certain, but one thing that appears to make a difference is how the base image is specified for the node pool VMS.  You can have the MSI and pod identity properly configured, but if the VM image is specified as shown below, the whole thing will fail, and it will present itself as an improperly configured MSI.

Does not work:
```json
    "hardwareProfile": {
      "vmSize": "Standard_D4s_v3"
    },
    "storageProfile": {
      "imageReference": {
        "id": "/subscriptions/109a5e88-712a-48ae-9078-9ca8b3c81345/resourceGroups/AKS-Ubuntu/providers/Microsoft.Compute/galleries/AKSUbuntu/images/1604/versions/2020.08.06"
```

Works as expected:
```json
    "hardwareProfile": {
      "vmSize": "Standard_D4s_v3"
    },
    "storageProfile": {
      "imageReference": {
        "publisher": "microsoft-aks",
        "offer": "aks",
        "sku": "aks-ubuntu-1604-202005",
        "version": "2020.05.31"
      },
```

I don't know how you get a node pool defined as in the "not working" example.  And since you can't deploy kubernetes < 1.16 any more, it might not be a valid point.  New deployments should use pod identity 1.6+.

**Comment by aramase** [comment]
@leeeennyy As @aullom1 posted, it is for the base images hosted in 
```
        "id": "/subscriptions/109a5e88-712a-48ae-9078-9ca8b3c81345/resourceGroups/AKS-Ubuntu/providers/Microsoft.Compute/galleries/AKSUbuntu/images/1604/versions/2020.08.06"
```

The cluster SP/managed identity doesn't have access to that subscription and aad-pod-identity < `1.6` release will fail because of insufficient permissions.

**Comment by leeeennyy** [comment]
Okay, sweet as @aullom1 @aramase I was just wondering because I still have some running on 1.16 of kubernetes, but I've started the process of upgrading pod identity and kubes version. Thanks for your help. 👍

**Comment by chenlonglab198** [comment]
May I know if there any downtime when we upgrade pod identity from 1.3 to 1.7 in the production. Thanks!
**current version:**
AKS version:1.17.11
MIC version:1.3
nmi:1.4

**Comment by chewong** [comment]
I think you should be able to achieve zero downtime with a rolling upgrade but there is going be some configuration update since your deployment version is pretty outdated.

**Comment by Yiyito11** [comment]
**_"Unfortunately you will need to upgrade pod identity to the latest releaese, which uses PATCH to assign / unassign identity from the underling VM / VMSS instead of CreateOrUpdate, which should get rid of the error"_**  Having the same issue Here, how you update the pod identityto lattest release? thank you.-

**Comment by chewong** [comment]
You will have to perform `helm upgrade` or simply one of `kubectl apply -f` from https://github.com/Azure/aad-pod-identity/releases/tag/v1.7.1. Note that there are multiple breaking changes between minor releases (see https://azure.github.io/aad-pod-identity/docs/#v17x-breaking-change)

**Comment by Yiyito11** [comment]
Yeah I deleted with kubectl delete -f and then recreated with that new version 1.7.1. Thanks anyway
